client/src_editor/Editor_Blocks.re

```diff
@@ -85,33 +85,46 @@ let make = (~blocks: array(block), _children) => {
                };
              }),
       })
-    | Block_Execute(blockId) =>
-      let block = state.blocks |. arrayFind(({b_id}) => b_id == blockId);
-      switch (block) {
-      | None => ReasonReact.NoUpdate
-      | Some(block) =>
-        switch (block.b_data) {
-        | B_Text(_) => ReasonReact.NoUpdate
-        | B_Code({bc_value}) =>
-          ReasonReact.SideEffects(
-            (
-              self =>
-                Js.Promise.(
-                  Editor_Worker.execute(. true, bc_value)
-                  |> then_(result => {
-                       let widgets =
-                         Editor_Blocks_Utils.executeRessultToWidget(result);
-                       resolve(
-                         self.send(Block_AddWidgets(blockId, widgets)),
-                       );
-                     })
-                  |> catch(error => resolve(Js.log(error)))
-                  |> ignore
-                )
-            ),
-          )
-        }
-      };
+    | Block_Execute(_blockId) =>
+      let allCodeBlocks =
+        Belt.Map.String.fromArray(
+          state.blocks
```

Comment:
This is really inefficient (3n). We can do this in a single loop like this:

```reason
module MS = Belt.Map.String;

state.blocks 
|. Belt.Array.reduceU(MS.empty, (. map, {b_id, b_data}) => {
  switch (b_data) {
  | B_Text(_) => map
  | B_Code({bc_value}) => map |. MS.set(b_id, bc_value)
  }
}))
```

As always, I didn't test this code :D